src/robotFactory.js

```diff
@@ -1,10 +1,69 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, position, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = {};
```

Comment:
```suggestion
    this.coords = {
      x: position.x ? position.x : 0,
      y: position.y ? position.y : 0,
    };
```
You may assign values directly here